ext/commonmarker/src/comrak_options.rs

```diff
@@ -0,0 +1,107 @@
+use comrak::ComrakOptions;
+
+use magnus::{class, r_hash::ForEach, Error, RHash, Symbol, Value};
+
+fn iterate_parse_options(comrak_options: &mut ComrakOptions, options_hash: RHash) {
+    options_hash.foreach(|key: Symbol, value: Value| {
+        if key.name().unwrap() == "smart" {
+            comrak_options.parse.smart = value.try_convert::<bool>()?;
+        }
+        if key.name().unwrap() == "default_info_string" {
+            comrak_options.parse.default_info_string = Some(value.try_convert::<String>().unwrap());
+        }
+        Ok(ForEach::Continue)
+    });
+}
+
+fn iterate_render_options(comrak_options: &mut ComrakOptions, options_hash: RHash) {
+    options_hash.foreach(|key: Symbol, value: Value| {
+        if key.name().unwrap() == "hardbreaks" {
+            comrak_options.render.hardbreaks = value.try_convert::<bool>()?;
+        }
+
+        if key.name().unwrap() == "github_pre_lang" {
+            comrak_options.render.github_pre_lang = value.try_convert::<bool>()?;
+        }
+
+        if key.name().unwrap() == "width" {
+            comrak_options.render.width = value.try_convert::<usize>()?;
+        }
+
+        if key.name().unwrap() == "unsafe_" {
+            comrak_options.render.unsafe_ = value.try_convert::<bool>()?;
+        }
+
+        if key.name().unwrap() == "escape" {
+            comrak_options.render.escape = value.try_convert::<bool>()?;
+        }
+
+        Ok(ForEach::Continue)
+    });
+}
+
+fn iterate_extension_options(comrak_options: &mut ComrakOptions, options_hash: RHash) {
+    options_hash.foreach(|key: Symbol, value: Value| {
+        if key.name().unwrap() == "strikethrough" {
```

Comment:
nit: this may read better as a match, and avoid some allocations:

```rust
match key.name() {
  Ok("strikethrough") => ...,
  Ok("tagfilter") => ...,
}
```